lib/ramble/ramble/workspace/workspace.py

```diff
@@ -1148,6 +1148,62 @@ def create_mirror(self, mirror_root):
         self.input_mirror_cache = ramble.caches.MirrorCache(self.input_mirror_path)
         self.software_mirror_cache = ramble.caches.MirrorCache(self.software_mirror_path)
 
+    def simplify(self):
+        # First drop unused experiment templates from app dict so environments aren't rendered
+        app_dict = ramble.config.config.get_config('applications',
+                                                   scope=self.ws_file_config_scope_name())
+
+        # Build experiment sets to determine which templates never get used
+        self.software_environments = \
+            ramble.software_environments.SoftwareEnvironments(self)
+        experiment_set = self.build_experiment_set()
+
+        # This uses experiments dict directly because all_experiments() filters out templates
```

Comment:
Seems like this would be better as a: `experiment_set.template_experiments()` iterator. Similar to what you did with packages and environments.

Reply:
Done